Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -50,19 +50,24 @@ class SIRT(Algorithm):
     upper : :obj:`float`, default = None
             Upper bound constraint, default value = :code:`-inf`.
     constraint : IndicatorBox function, default = None
```

Comment:
In docs, in the first line, 52, we usually write the `class` that describes the parameter, e.g.,
`data : DataContainer`, `lower : obj:`float`. Also, in the docs a url is created and the user can click and go to `DataContainer` or `LinearOperator`. 
And there is no `indicator` function to refer to.

Reply:
You are right, let us keep it with IndicatorBox for now then. But in fact, what can we do with IndicatorBox at present that we can't with lower/upper?

I think (not in this PR) we should add a base IndicatorFunction class from which IndicatorBox is subclassed. And add at least one more indicator function of some sort.

Then also we should expand IndicatorBox to take in upper and lower bounds as a vector instead of a scalar to allow different constraint across the pixels. In principle this could also be done for lower and upper arguments of SIRT.

Reply:
"""But in fact, what can we do with IndicatorBox at present that we can't with lower/upper?"""

If `constraint` is `None` then we check for the `lower` and `upper` args. Then, we create and `IndicatorBox` to use as `proximal` step. If `constraint` is not `None` the a proximal method should be implemented.

```
    Note 
    ----
    
    If :code:`constraint` is not passed, then :code:`lower` and :code:`upper` are looked at and an :class:`.IndicatorBox`
    function is created.
    If :code:`constraint` is passed, :code:`proximal` method is required to be implemented.
    If :math:`M = \frac{1}{A*\mathbb{1}}`, :math:`D = \frac{1}{A^{T}\mathbb{1}}` contain :code:`NaN`, :math:`\pm\inf`
    they are replaced by :math:`1`.
```

Totally agree but not in this PR. There are many proximal method that implement directly the Indicator of L\infty, or indicator of a point etc.